components/measure-upload/MeasureRepositoryUpload.tsx

```diff
@@ -0,0 +1,235 @@
+  const [isLoadingIds, setIsLoadingIds] = useState(false);
+  const [isLoadingPackage, setIsLoadingPackage] = useState<PACKAGE_STATES>(PACKAGE_STATES.NONE);
+  const [measureSelectIsOpen, setMeasureSelectIsOpen] = useState(false);
+
+  const [{ measureRepositoryUrl, isFile, selectedMeasureId, displayMap }, setMeasureBundle] =
+    useRecoilState(measureBundleState);
+  const setMeasurementPeriod = useSetRecoilState(measurementPeriodState);
+
+  const idSelectData = useRecoilValue(displayMapToSelectDataState);
+
+  useEffect(() => {
+    if (isFile) {
+      setMeasureSelectIsOpen(false);
+      setIsLoadingPackage(PACKAGE_STATES.NONE);
+    }
+  }, [isFile]);
+
+  const rejectUpload = (
+    message: string | string[],
+    attemptedBundleDisplay: string | null,
+    isValueSetMissingError = false
+  ) => {
+    logError({
+      id: uuidv4(),
+      message,
+      timestamp: new Date().toISOString(),
+      attemptedBundleDisplay,
+      isValueSetMissingError
+    });
+    showNotification({
+      icon: <IconAlertCircle />,
+      title: 'Bundle upload failed',
+      message: 'There was an issue uploading your bundle. Please correct the issues listed below.',
+      color: 'red'
+    });
+    setIsLoadingPackage(PACKAGE_STATES.FAIL);
+  };
+
+  /*
+    Handles updating of the Measure Repository Service url field and associated state variables
+  */
+  function handleMrsUrlChange(event: React.ChangeEvent<HTMLInputElement>) {
+    if (measureSelectIsOpen) {
+      setMeasureSelectIsOpen(false);
+    }
+    setMeasureBundle(mb => ({
+      ...mb,
+      measureRepositoryUrl: event.currentTarget.value,
+      selectedMeasureId: null,
+      displayMap: {},
+      ...(!mb.isFile ? { content: null } : undefined)
+    }));
+    setIsLoadingPackage(PACKAGE_STATES.NONE);
+  }
+
+  /*
+    Takes the entries from a Measure retrieval response and creates a map from Measure id
+    to Measure display label for Measure select component
+  */
+  function createMeasureDisplayMap(
+    displayMap: Record<string, string>,
+    { resource: measure }: { resource: fhir4.Measure }
+  ) {
+    let label: string;
+    const officialIdentifier = measure.identifier?.find(
+      identifier =>
+        identifier.use === 'official' && identifier.system === 'http://hl7.org/fhir/cqi/ecqm/Measure/Identifier/cms'
+    );
+    if (officialIdentifier?.value) {
+      label = measure.version ? `${officialIdentifier.value}|${measure.version}` : officialIdentifier.value;
+    } else if (measure.name) {
+      label = measure.version ? `${measure.name}|${measure.version}` : measure.name;
+    } else {
+      label = measure.id as string;
+    }
+    displayMap[measure.id as string] = label;
+    return displayMap;
+  }
+
+  /*
+    Sends a `GET` request to the `/Measure` endpoint of the specified Measure Repository Service.
+    Parses and stores the measureIds if successful
+  */
+  async function retrieveMeasures() {
```

Comment:
When a mrs doesn't have any measures, the measure dropdown is disabled which is probably intuitive enough for the user but maybe we could add an alert that says that there are no measures in the provided measure repository service url?

Reply:
I like that idea - was going to say the same thing